docs/graphs/graph-concepts/graphs-and-indexes.md

```diff
@@ -0,0 +1,105 @@
+
+### Edge-Centric Indexes
+
+Edge-centric indexes optimize queries involving filtering or sorting based on edge properties. In Macrometa, create a "hash", "skiplist", or "persistent" index on an edge collection to expedite queries filtering edges by specific attributes. For example, if querying relationships in a social network graph by a "relationship type" attribute is common, a "hash" index on the relationship type can enhance query performance.
+
+### Improving Traversal Performance
+
+Traversal performance can be impacted by the choice of indexes on vertex and edge collections. Proper indexing can quickly locate traversal starting points and navigate the graph efficiently. Analyze your application's specific traversal patterns to determine the most beneficial indexes.
+
+### Balancing Indexing
+
+When creating indexes, balance the benefits of indexing with the overhead of maintaining them. Indexes improve query performance but consume storage space and require extra maintenance during data modifications. Create indexes selectively, concentrating on the properties and relationships frequently used in your application's queries.
+
+## Choosing the Right Index Type for Graph Performance
+
+Selecting the appropriate index type for your specific use case and query patterns is crucial for enhancing graph performance. Consider the following index types for graph data in Macrometa.
+
+### Hash Index
+
+A hash index is ideal for equality lookups on single attributes. It offers fast document access based on the indexed attribute but does not support range queries or sorting. If your graph queries involve filtering vertices or edges by a specific attribute's exact value, then a hash index can be useful.
+
+**Use case**: Searching for users by their email address or finding edges with a specific relationship type.
+
+### Skiplist Index
+
+A skiplist index is suitable for single or multiple attributes, supporting both equality lookups and range queries. It also enables sorting by the indexed attribute(s). If your graph queries involve filtering or sorting vertices or edges based on one or multiple attributes' values, particularly when range queries are involved, a skiplist index can be helpful.
+
+**Use case**: Locating posts within a specific date range or sorting edges by a weight attribute.
+
+### Persistent Index
+
+A persistent index is similar to a skiplist index in terms of supported queries, but it is stored on disk rather than in memory. This index type is useful for large datasets that do not fit in memory, but it might have slightly higher access latency compared to in-memory indexes. If your graph queries involve large datasets and require filtering or sorting based on attribute values, then a persistent index might be an appropriate choice.
```

Comment:
the concepts "persistent" and "skiplist" are orthogonal. The following provides some context (although this is for a newer version of arango and the details may be inaccurate): https://www.arangodb.com/docs/stable/indexing-working-with-indexes.html

Persistent index means an index which is stored on disk. Skiplist index means an index which preserves the order of indexed items (as opposed to hash index).

Reply:
Okay...so how do I change this text?